### DISCUSSION TITLE
Code blocks and Copy to Clipboard questions

### DISCUSSION THREAD

**Discussion by ciaransweet** [mkdocs/mkdocs#2522] [Q&A]
Hi folks 👋 

Not issues as such, I'm just not sure where best to ask these kind of questions!

* For a code block (```) - is there a way to specify content *not* to copy within the block? I.E virtual env names at the start of a terminal output (`(venv) $ <the command>`) - I only want to copy `the command`

* For a code block (```) - is there a way to completely disable `Copy to Clipboard`? - For outputs of commands like `--help`, it doesn't seem useful to be able to copy this

Many thanks! Apologies if this isn't the place!

**Comment by facelessuser** [comment]
Are you using a specific theme? I don't think the default MkDocs theme offers "copy to clipboard" functionality.

**Reply by ciaransweet** [reply]
@facelessuser great question! I'm new to the project so looking at it, we're using `material`:

```yaml
theme:
  name: material
```

**Reply by facelessuser** [reply]
Yeah, that is a Material specific feature. I also use Material. And as for your question, there is no way to exclude certain content. It is a simple copy the code block feature.

**Reply by ciaransweet** [reply]
So given I cannot exclude certain content - is there a way to disable the copy as a whole for a block?

**Reply by facelessuser** [reply]
Maybe, I don't recall. This is really a question for the Material theme, but maybe @squidfunk can answer it here?

**Reply by squidfunk** [reply]
No, and yes/no.

1. In order to omit certain characters from the start, middle or end, we would need to add custom handling logic for each language. From the top of my head, I can't imagine this working in any reliable cross-language way.

2. https://github.com/squidfunk/mkdocs-material/issues/2418#issuecomment-794571375

**Reply by ciaransweet** [reply]
@squidfunk thanks for the clarification!

**Comment by oprypin** [comment]
Relying on syntax highlighting by Pygments, you can make the prompts not selectable by adding custom CSS as such:

```css
.highlight .gp { /* Generic.Prompt */
  user-select: none;
}
```

You can see this in action at
https://mkdocstrings.github.io/crystal/

And apparently that also affects the "copy to clipboard" button, at least in Firefox for me.

Similar might be possible with highlightjs.

**Reply by squidfunk** [reply]
That is super clever! Haven’t thought of this approach. We should add this to Material for MkDocs. If somebody wants to go head, PR appreciated!

**Reply by oprypin** [reply]
@squidfunk I think forcing this setting would be an incompatible change, and generally it can have unforeseen effects, as multiple highlighting languages are affected. But would be a great addition to documentation.

**Reply by squidfunk** [reply]
I agree. However, if the Highlight extension would annotate code blocks with their language, we could scope it to shell-like languages, like this:

``` css
.bash.highlight .gp {
  user-select: none;
}
```

BTW, I just tested and it doesn't work on iOS, but I guess it's a "nice to have" anyways.

**Reply by ciaransweet** [reply]
@oprypin sorry to raise this from the dead.

I just added the CSS block you linked and it works a charm, hooray! *Edit* On chrome anyway 

Cheekily, could I ask you to elaborate more on your original comment?

```
Relying on syntax highlighting by Pygments, you can make the prompts not selectable by adding custom CSS as such:
```

Where in their documentation/examples you may have, is this an obvious thing to do?

It still appears to be magic to me 😅
